src/lib.rs

```diff
@@ -190,313 +182,415 @@ pub struct AstarteDeviceDataEvent {
     pub data: Aggregation,
 }
 
-impl<S> AstarteDeviceSdk<S>
-where
-    S: PropertyStore,
-{
-    /// Create a new instance of the Astarte Device SDK.
+#[async_trait]
+pub trait Device {
+    // ------------------------------------------------------------------------
+    // object types
+    // ------------------------------------------------------------------------
+
+    /// Send an object datastreamy on an interface, with an explicit timestamp.
     ///
     /// ```no_run
-    /// use astarte_device_sdk::{AstarteDeviceSdk, options::AstarteOptions};
+    /// # use astarte_device_sdk::{AstarteDeviceSdk, Device, builder::{DeviceBuilder, MqttConfig}, AstarteAggregate};
+    /// #[cfg(not(feature = "derive"))]
+    /// use astarte_device_sdk_derive::AstarteAggregate;
+    /// use chrono::{TimeZone, Utc};
+    ///
+    /// #[derive(AstarteAggregate)]
+    /// struct TestObject {
+    ///     endpoint1: f64,
+    ///     endpoint2: bool,
+    /// }
     ///
     /// #[tokio::main]
     /// async fn main() {
-    ///     let sdk_options = AstarteOptions::new("", "", "", "")
-    ///     .interface_directory("")
-    ///     .unwrap()
-    ///     .ignore_ssl_errors();
+    ///     let mqtt_config = MqttConfig::new("_", "_", "_", "_");
```

Comment:
Change the value of the string in realm_id, device_id, credential_secret, and pairing_url